domains/Order.js

```diff
@@ -0,0 +1,51 @@
+import MENU_BOARD from '../constants/menu.js';
+import OrderValidation from './OrderValidation.js';
+
+class Order {
+  #menu;
+
+  constructor(menu) {
+    this.#menu = menu;
+  }
+
+  static validate(menu) {
+    const orderValidation = new OrderValidation(menu);
+
+    if (
+      !orderValidation.checkIsOrderInForm() ||
+      !orderValidation.checkInMenu() ||
+      !orderValidation.checkIsMenuCountValid() ||
+      orderValidation.checkIsMenuDuplicated() ||
+      !orderValidation.checkIsTotalCountValid() ||
+      orderValidation.checkIsAllBeverage()
+    ) {
+      return false;
+    }
+    return true;
+  }
+
+  createOrderBoard() {
+    const orderBoard = [];
+    const eachMenu = this.#menu.map((item) => item.split('-'));
+    eachMenu.forEach((each) => {
+      orderBoard.push({ name: each[0], amount: Number(each[1]) });
+    });
```

Comment:
```suggestion
    const orderBoard = eachMenu.map((each) => ({ name: each[0], amount: Number(each[1]) }));
```
이런 식으로 map을 이용하면 push 메소드 사용 없이 불변성을 유지할 수 있을 것 같아요!

Reply:
이런 방법도 있었네요.. !!

map 메소드를 더 자주 사용해 봐야 겠습니다 !

감사합니다 !!